crates/query-engine/translation/src/translation/query/values.rs

```diff
@@ -38,6 +38,25 @@ pub fn translate_json_value(
                 r#type: type_to_ast_scalar_type(r#type),
             })
         }
+        // For composite types we use the `jsonb_populate_record` function to build the composite
+        // value rather than constructing the value field-by-field in SQL.
+        (serde_json::Value::Object(_obj), database::Type::CompositeType(_type_name)) => {
+            let stringified = serde_json::to_string(value)
```

Comment:
Worth considering passing a serde_json::Value instead of a string here.

Reply:
Oh! I didn't know we could!